Sources/CocoaLumberjack/DDLog.m

```diff
@@ -1037,10 +1037,20 @@ - (instancetype)initWithMessage:(NSString *)message
     return self;
 }
 
+static __inline__ __attribute__((__always_inline__)) BOOL _nullable_strings_equal(NSString* _Nullable lhs, NSString* _Nullable rhs)
+{
+    if (lhs == nil) {
+        if (rhs == nil)
+            return YES;
+    } else if (rhs != nil && [lhs isEqualToString:rhs])
+        return YES;
+    return NO;
+}
+
 - (BOOL)isEqual:(id)other {
     if (other == self) {
         return YES;
-    } else if (![super isEqual:other] || ![other isKindOfClass:[self class]]) {
```

Comment:
Can you add a comment why `super` isn't called here? Because overridden methods that do not call super is something that always catches my eye...

Reply:
Also, we could consider adding a `[super isEqual: other]` check right at the beginning. If super says we're equal, then we're equal. I'd still to the pointer check, in case `NSObject` ever decides to do something more.

Reply:
See https://stackoverflow.com/questions/36593038/confused-about-the-default-isequal-and-hash-implements